### DISCUSSION TITLE
File upload issue: When uploading multiple files some files are missing

### DISCUSSION THREAD

**Discussion by uma-ac** [filamentphp/filament#13305] [General]
When I upload multiple file one or two files are left behind. 
I can see that those files are in livewire-tmp folder. 

how to solve this issue.
![FileUploadMissingFile](https://github.com/filamentphp/filament/assets/121853633/d77938cc-2e63-41c5-96a2-478299b0d554)
<img width="744" alt="FileUploadMissingFile_temp" src="https://github.com/filamentphp/filament/assets/121853633/bb86b54a-56f9-4db0-9dc0-958dc5d2ef19">
Config : 
`  'temporary_file_upload' => [
        'disk' => 'tmp',        // Example: 'local', 's3'              | Default: 'default'
        "rules" => "file|mimes:pdf,jpg,jpeg,png,doc,docx,svg,mp4,mp3 | max:100000",
        // 'rules' => 'max:100000',       // Example: ['file', 'mimes:png,jpg']  | Default: ['required', 'file', 'max:12288'] (12MB)
        'directory' => null,   // Example: 'tmp'                      | Default: 'livewire-tmp'
        'middleware' => null,  // Example: 'throttle:5,1'             | Default: 'throttle:60,1'
        'preview_mimes' => [   // Supported file types for temporary pre-signed file URLs...
            'png', 'gif', 'bmp', 'svg', 'wav', 'mp4',
            'mov', 'avi', 'wmv', 'mp3', 'm4a',
            'jpg', 'jpeg', 'mpga', 'webp', 'wma', 'pdf'
        ],
        'max_upload_time' => 5, // Max duration (in minutes) before an upload is invalidated...
    ],
`
`FileUpload::make('attachments')
                        ->columnSpan(12)
                        ->multiple()
                        ->maxSize(100000)
                        ->disk('uploads')
                        ->visibility('private')
                        ->directory('application_files')
                        ->getUploadedFileNameForStorageUsing(function ($file): string {
                            $fileName = str($file->getClientOriginalName());

                            $fileName_str = preg_replace('/[^a-zA-Z0-9.]/','',iconv('UTF-8', 'ASCII//TRANSLIT',  $fileName));
                            return (string) str($fileName_str)->prepend(now() . '-');
                        })
                        ->downloadable()
                        ->rules([ new MalwareFile] )
                        ->label(__('application.resource.application_documents')),
`

`Section::make('')
                    ->relationship('result_files')
                    ->schema([
                    FileUpload::make('attachments')
                        ->columnSpan(12)
                        ->multiple()
                        ->maxSize(100000)
                        ->disk('uploads')
                        ->directory('result_files')
                        ->acceptedFileTypes(["application/pdf"])
                        ->getUploadedFileNameForStorageUsing(function ($file): string {
                            $fileName = str($file->getClientOriginalName());
                            $fileName_str = preg_replace('/[^a-zA-Z0-9.]/','',iconv('UTF-8', 'ASCII//TRANSLIT',  $fileName));
                            return (string) str($fileName_str)->prepend(now() . '-');
                        })
                        ->downloadable()
                        ->rules([ new MalwareFile] )
                        ->label(__('application.resource.result_documents'))
                ]),`

**Comment by PegeDev** [comment]
any update for this issue ??

**Comment by uma-ac** [comment]
I dont think so.

**Comment by karamqubsi** [comment]
I faced this problem today, 

In short: if you have this behind a load balancer and you have more than one server / container being served through the LB, make sure you enable sticky sessions in your LB, otherwise files will be uploaded in 2 or more servers ( to livewire temp folders ) and only one server will continue the process in saving the data to db, and in that one server few of those files may not exists,

or make sure temp livewire folder is shared between all servers.  

For me sticky sessions did the trick.
